### PR TITLE
docs: expand the WSL guide

### PR DESCRIPTION
**Pull request by leonfeng**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This PR updates parts of the WSL guide and includes several tips I found while troubleshooting my own WSL setup. Thank you for reviewing my PR!

### CODE REVIEW

File: docs/how-to-setup-wsl.md [hunk 1]
```diff
@@ -73,10 +75,10 @@ Once you have configured Docker Desktop to work with WSL2, follow these steps to
 
 1. Launch a new Ubuntu-18.04 terminal
 
-2. Pull `MongoDB 4.0.x` from Docker Hub
+2. Pull `MongoDB 4.4.x` from Docker Hub
```
Comment (raisedadead) [inline docs/how-to-setup-wsl.md:78]: We are on MongoDB 6 and will soon move to 7, too. Let's point them to system requirements on the main guide to avoid updating this all the time.

File: docs/how-to-setup-wsl.md [hunk 2]
```diff
@@ -73,10 +75,10 @@ Once you have configured Docker Desktop to work with WSL2, follow these steps to
...
 
    ```console
-   docker pull mongo:4.0
+   docker pull mongo:4.4
```
Comment (raisedadead) [inline docs/how-to-setup-wsl.md:81]: See earlier comment.

File: docs/how-to-setup-wsl.md [hunk 3]
```diff
@@ -87,7 +89,7 @@ Once you have configured Docker Desktop to work with WSL2, follow these steps to
      -p 27017:27017 \
      --name mongodb \
      --restart unless-stopped \
-     -d mongo:4.0
+     -d mongo:4.4
```
Comment (raisedadead) [inline docs/how-to-setup-wsl.md:92]: See earlier comment.

File: docs/how-to-setup-wsl.md [hunk 4]
```diff
@@ -104,12 +106,12 @@ nvm install --lts
...
 # nvm install <version>
 
-nvm install 14
+nvm install 20.9
 
 # Usage
 # nvm use <version>
 
-nvm use 12
+nvm use 20.9
```
Comment (raisedadead) [inline docs/how-to-setup-wsl.md:114]: Let's remove this because clearly, users are skipping over the fact the `--lts` is a better option. We can point users to nvm usage on the nvm docs.

File: docs/how-to-setup-wsl.md [hunk 5]
```diff
@@ -126,6 +128,65 @@ Now that you have installed the pre-requisites, follow [our local setup guide](h
...
+   >
+   > The following tips were collected from across the web and have not gone through vigorous testing. Your mileage may vary.
+
+### Adjust processer scheduling for background services
+
+This may reduce incidents of Docker containers crashing due to lack of resources.
+
+Open the System Properties control panel by pressing <kbd>Win + R</kbd> and entering `sysdm.cpl`
+
+![Run dialog](images/wsl/run-sysdm.png)
```
Comment (raisedadead) [inline docs/how-to-setup-wsl.md:143]: Wrap images in a collapsed details view. 

Example:
https://github.com/freeCodeCamp/freeCodeCamp/blob/5c5889dd3a3cd6ad2eb8cc1fef037e64e5c15abb/docs/how-to-setup-freecodecamp-locally.md?plain=1#L127-L133

Note that you must do this in a separate PR to use the `` tag and upload the images first.

**Review by raisedadead — CHANGES_REQUESTED** [review]
Nice work.

File: docs/how-to-setup-wsl.md [hunk 6]
```diff
@@ -126,6 +128,65 @@ Now that you have installed the pre-requisites, follow [our local setup guide](h
...
+```ini
+# Settings apply across all Linux distros running on WSL 2
+[wsl2]
+
+# Limits VM memory to use no more than 12 GB, this can be set as whole numbers using GB or MB
+memory=12GB
+
+# Sets amount of swap storage space to 8GB, default is 25% of available RAM
+swap=8GB
+```
```
Comment (ShaunSHamilton) [inline docs/how-to-setup-wsl.md:176]: For this, I suggest just mentioning default memory might be an issue and pointing to the docs: https://learn.microsoft.com/en-us/windows/wsl/wsl-config#configuration-setting-for-wslconfig

There are some gotchas the docs mention that Campers should be aware of before they blindly copy `memory=12GB` in their own config.

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thank you for taking the time to contribute

File: docs/how-to-setup-wsl.md (hunk 5 above)
Comment (leonfeng) [reply docs/how-to-setup-wsl.md:143]: You mean like this? https://github.com/freeCodeCamp/freeCodeCamp/pull/52239

Comment (raisedadead) [reply docs/how-to-setup-wsl.md:135]: Yes, that is fine.

**Review by raisedadead — APPROVED** [review]

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Comment by ShaunSHamilton** [comment]
Thank you for this PR, @leonfeng 🚀
